src/Symfony/Component/DependencyInjection/Dumper/XmlDumper.php

```diff
@@ -298,6 +298,10 @@ private function convertParameters(array $parameters, $type, \DOMElement $parent
                 $element->setAttribute('type', 'expression');
                 $text = $this->document->createTextNode(self::phpToXml((string) $value));
                 $element->appendChild($text);
+            } elseif (is_string($value) && \in_array(preg_match('/[^\s\P{Cc}]/u', $value), [false, 1], true)) {
```

Comment:
Why do we need to check for both `false` and `1`? Do we cover both values with the current test suite?

Reply:
`false` is returned when value contains non-utf8 characters. `1` is returned when a control character is present in string. You're right a case is missing in current test suite.